client.go

```diff
@@ -15,18 +15,20 @@ type CustomHTTPClient struct {
 	dedupeHashTable     *sync.Map
 	dedupeOptions       DedupeOptions
 	skipHTTPStatusCodes []int
+	errorChannel        chan error
 	WARCTempDir         string
 }
 
 func (c *CustomHTTPClient) Close() error {
 	c.WaitGroup.Wait()
 	c.CloseIdleConnections()
 	close(c.WARCWriter)
+	close(c.errorChannel)
```

Comment:
Shouldn't the close happen AFTER the warc writer is fully finished? Maybe I'm reading the code wrong.